src/tiled/mainwindow.cpp

```diff
@@ -816,6 +817,25 @@ bool MainWindow::saveFileAs()
     return saveFile(fileName);
 }
 
+bool MainWindow::saveAll()
+{
+    foreach (MapDocument *mapDoc, mDocumentManager->documents()) {
+        QString fileName(mapDoc->fileName());
+        if (fileName.isEmpty())
```

Comment:
This is the case for new documents that were not saved yet, right? I think in this case, we should probably switch the current document and pop up a dialog to choose the file name (ie. call `saveFileAs`).